jerry-main/main-unix-snapshot.c

```diff
@@ -73,6 +93,237 @@ read_file (uint8_t *input_pos_p, /**< next position in the input buffer */
   return bytes_read;
 } /* read_file */
 
+/** Generate command line option IDs
+ *
+ */
+typedef enum
+{
+  OPT_GENERATE_HELP,
+  OPT_GENERATE_CONTEXT,
+  OPT_GENERATE_SHOW_OP,
+  OPT_GENERATE_LITERAL_LIST,
+  OPT_GENERATE_LITERAL_C,
+  OPT_GENERATE_OUT,
+} generate_opt_id_t;
+
+/**
+ * Generate command line options
+ */
+static const cli_opt_t generate_opts[] =
+{
+  CLI_OPT_DEF (.id = OPT_GENERATE_HELP, .opt = "h", .longopt = "help",
+               .help = "print this help and exit"),
+  CLI_OPT_DEF (.id = OPT_GENERATE_SHOW_OP, .longopt = "show-opcodes",
+               .help = "print generated opcodes"),
+  CLI_OPT_DEF (.id = OPT_GENERATE_CONTEXT, .opt = "c", .longopt = "context",
+               .help = "specify the execution context of the snapshot: "
+                       "global or eval (default: global).", .meta = "MODE"),
+  CLI_OPT_DEF (.id = OPT_GENERATE_LITERAL_LIST, .longopt = "save-literals-list-format",
+               .meta = "FILE",
+               .help = "export literals found in parsed JS input (in list format)"),
+  CLI_OPT_DEF (.id = OPT_GENERATE_LITERAL_C, .longopt = "save-literals-c-format",
+               .meta = "FILE",
+               .help = "export literals found in parsed JS input (in C source format)"),
```

Comment:
the original `--save-snapshot-for-{global,eval}` options have already been rewritten to a new `--context={global,eval}` format. wouldn't it make sense to have something similar for `--save-literals-{c,list}-format`, too? e.g., as simple as `--save-literals={c,list}`.

Reply:
the logic is different, for --save-literals* you need to specify the output file as argument. This way you can generate the literal file and also the snapshot.

Reply:
oh that's right